src/core/backbone_router/local.cpp

```diff
@@ -0,0 +1,246 @@
+    // refresh its dataset.
+    VerifyOrExit(aForce || !Get<BackboneRouter::Leader>().HasPrimary() || mState == OT_BACKBONE_ROUTER_STATE_PRIMARY);
+
+    serverData.SetSequenceNumber(mSequenceNumber);
+    serverData.SetReregistrationDelay(mReregistrationDelay);
+    serverData.SetMlrTimeout(mMlrTimeout);
+
+    error = Get<NetworkData::Local>().AddService(THREAD_ENTERPRISE_NUMBER, &serviceData, sizeof(serviceData), true,
+                                                 reinterpret_cast<const uint8_t *>(&serverData), sizeof(serverData));
+exit:
+    if (error == OT_ERROR_NONE)
+    {
+        mIsServiceAdded = true;
+        otLogInfoNetData("BBR Service added: seqno (%d), delay (%ds), timeout (%ds)", mSequenceNumber,
+                         mReregistrationDelay, mMlrTimeout);
+        error = Get<NetworkData::Local>().SendServerDataNotification();
+    }
+
+    otLogInfoNetData("Add BBR Service: %s", otThreadErrorToString(error));
+    return error;
+}
+
+otError Local::RemoveService(void)
+{
+    otError error       = OT_ERROR_NONE;
+    uint8_t serviceData = NetworkData::ServiceTlv::kServiceDataBackboneRouter;
+
+    error = Get<NetworkData::Local>().RemoveService(THREAD_ENTERPRISE_NUMBER, &serviceData, sizeof(serviceData));
+    SuccessOrExit(error);
+
+    error = Get<NetworkData::Local>().SendServerDataNotification();
+
+exit:
+    otLogInfoNetData("Remove BBR Service %s", otThreadErrorToString(error));
+    return error;
+}
+
+otError Local::AddBackboneRouterPrimaryAloc(void)
+{
+    memcpy(&mBackboneRouterPrimaryAloc.GetAddress(), &Get<Mle::MleRouter>().GetMeshLocal16(), 14);
+    mBackboneRouterPrimaryAloc.GetAddress().mFields.m16[7] = HostSwap16(Mle::kAloc16BackboneRouterPrimary);
+    return Get<ThreadNetif>().AddUnicastAddress(mBackboneRouterPrimaryAloc);
+}
+
+void Local::NotifyBackboneRouterPrimaryUpdate(Leader::State aState, const BackboneRouterConfig &aConfig)
+{
+    OT_UNUSED_VARIABLE(aConfig);
+
+    switch (aState)
+    {
+    case Leader::kStateToTriggerRereg:
+        // fall through
+    case Leader::kStateAdded:
+        if (Get<BackboneRouter::Leader>().GetServer16() == Get<Mle::MleRouter>().GetRloc16())
+        {
+            if (mState == OT_BACKBONE_ROUTER_STATE_SECONDARY)
+            {
+                if (!mIsServiceAdded)
+                {
+                    // Here original PBBR restores its Backbone Rouetr Service from Thread Network,
+                    // Intentionally skips the state update as PBBR will refresh its service.
+                    mSequenceNumber      = aConfig.mSequenceNumber + 1;
+                    mReregistrationDelay = aConfig.mReregistrationDelay;
+                    mMlrTimeout          = aConfig.mMlrTimeout;
+                    AddService(true /* Force registration to refresh and restore Primary state */);
+                }
+                else
+                {
+                    AddBackboneRouterPrimaryAloc();
```

Comment:
The `local` and `leader` in namespace `BackboneRouter` are some kind of `cache` components, 
`local` cache for local bbr dataset with additionally logic to decide whether or not contribute to local server data as bbr service should be unique in general, the `leader` is to cache the primary backbone router service per network data update as the status of change of it may cause extra actions for other components.  The consistency check is left to the network_data_local in core per network data update or per MLE advertisement from neighbor.

 if local SeqNum is in local server data, the core will be responsible for the consistency check and register it if inconsistent. Why needs increase SeqNum and force registration?